### ISSUE TITLE
Bug with yarn version 3: impossible to execute

### ISSUE THREAD

**Issue by dsumac** [gunar/shellcheck#33]
Hello.
Firstly, thank you for the work on this wrapper.

Today, we have seen a problem at execution.

Here is the stackTrace: 

```
/Users/.../testProject/node_modules/shellcheck/shellcheck-stable/shellcheck:1
����


SyntaxError: Invalid or unexpected token
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1033:15)
    at Module._compile (node:internal/modules/cjs/loader:1069:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47
```

After a lot of investigations, we have tested with a new project in these configurations:

- npm version 8.5.5: OK
- yarn version 1.22.18: OK
- yarn version 3.2.2 (in [zero install mode](https://yarnpkg.com/features/zero-installs)): **KO**

For the moment, we have founded a dirty but functionnal workaround, in the scripts use a path from node_modules:

` "shellcheck": "node_modules/shellcheck/shellcheck-stable/shellcheck *.sh",`

Could you fix this bug ?

**Comment by gunar** [comment]
Can you post the steps to reproduce? Do you get this error when installing or when trying to run shellcheck?

**Comment by dsumac** [comment]
The steps are:

- creating new project from scratch with yarn following [the install guide ](https://yarnpkg.com/getting-started/install) (you can also have an installation of yarn 1 and make an upgrade with `yarn set version stable` )
- create a script file at project root `touch testScript.sh`
- create a script in `package.json`  "lint:shellcheck": "shellcheck testScript.sh"
- run the script via `yarn lint:shellcheck`
- then the previous error is displayed

**Comment by dsumac** [comment]
Are the steps clear enough?
Were you able to reproduce the problem?

**Comment by gunar** [comment]
Steps are clear. I don't use `yarn` zero-install myself so I can't say that l'll have time to try and reproduce it in the near future but if you (or anyone) find a fix I'm happy to merge the PR.

If you really want/need my help, what you could do to help me help you is provide a `Dockerfile` that reproduces the bug. That'd make it easier for me to find and fix the bug Mr. Camus. (If indeed there is a solution that works with Yarn Zero-Install).

**Comment by carlocorradini** [comment]
Please, see PR #34

**Comment by gunar** [comment]
This will soon be released in #34.
